sql/analyzer/resolve_orderby.go

```diff
@@ -57,7 +59,15 @@ func resolveOrderBy(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, error)
 		// below its child.
 		if len(colsFromChild) == 0 && len(missingCols) > 0 {
 			a.Log("pushing down sort, missing columns: %s", strings.Join(missingCols, ", "))
-			return pushSortDown(sort)
+			newNode, err := pushSortDown(sort)
+			if err != nil {
+				return newNode, err
+			}
+			_, ok := newNode.(*plan.ResolvedTable)
```

Comment:
What happens if you remove this check? Does the execution plan not catch the unresolved columns in some other way later on?

Reply:
Correct, the ORDER BY doesn't fail or anything, it essentially just gets ignored. Comparing the behavior with MySQL, it's supposed to error out, so this is that check.